distributed/comm/tcp.py

```diff
@@ -473,8 +473,12 @@ def get_address_host(self, loc):
     def get_address_host_port(self, loc):
         return parse_host_port(loc)
 
-    def resolve_address(self, loc):
+    def resolve_address(self, loc, clear_cache=False):
         host, port = parse_host_port(loc)
+        if clear_cache:
+            result = ensure_ip(host)
+            ensure_ip.clear_cache(host)
+            return unparse_host_port(result, port)
```

Comment:
Maybe the thing to do here is to invalidate the cache before calling `ensure_ip`.  That might result in slightly simpler code.